packages/webdriverio/src/commands/browser/switchFrame.ts

```diff
@@ -265,6 +265,33 @@ export async function switchFrame (
         await element.waitForExist({
             timeoutMsg: `Can't switch to frame with selector ${element.selector} because it doesn't exist`
         })
+
+        // If this is an element from a collection (has index property)
+        if (element.index !== undefined && element.selector) {
+            try {
+                // CSS :nth-child and XPath position() are 1-based
+                const index = element.index + 1
+                let selector: string
+
+                if (element.locator?.type === 'xpath') {
+                    const baseXpath = element.locator.value || element.selector
+                    selector = `(${baseXpath})[${index}]`
+                } else {
+                    selector = `${element.selector}:nth-child(${index})`
+                }
+
+                const specificElement = this.$(selector)
+                await specificElement.waitForExist({
+                    timeoutMsg: `Can't switch to frame at index ${element.index} because it doesn't exist`
+                })
+
+                const frameElement = await specificElement.getElement()
+                return switchToFrameUsingElement(this, frameElement)
+            } catch (error) {
+                log.warn(`Failed to locate frame element at index ${element.index}: ${(error as Error).message}`)
+            }
```

Comment:
What is the reason to wrap this into a `try/catch` shouldn't we throw an error if we fail to switch the frame?

Reply:
HI @christian-bromann ,

Thank you for the review.  looking into issue https://github.com/webdriverio/webdriverio/issues/14418

I see `context.getElement()` is picking the first element, ignoring the indexes for element passes and selects to default first one

https://github.com/webdriverio/webdriverio/blob/20ebd75b6ab890dddad0f383f84b2c970c67c712/packages/webdriverio/src/commands/browser/switchFrame.ts#L264

I thought we can add check before/after as fallback mechanism get the correct element before switch

Please suggest on this!